test/Microsoft.AspNet.Mvc.Core.Test/DefaultAssemblyProviderTests.cs

```diff
@@ -164,5 +205,100 @@ public NullAssemblyProvider(ILibraryManager libraryManager) : base(libraryManage
             {
             }
         }
+
+        private class MvcAssembliesTestingProvider : DefaultAssemblyProvider
+        {
+            private static readonly ILibraryManager _libraryManager = GetLibraryManager();
+            private static readonly string _mvcName = "Microsoft.AspNet.Mvc";
+            private static HashSet<string> _mvcReferenceAssemblies;
+
+            public MvcAssembliesTestingProvider() : base(_libraryManager)
+            { }
+
+            public HashSet<string> LoadableReferenceAssemblies
+            {
+                get
+                {
+                    if (_mvcReferenceAssemblies != null)
+                    {
+                        return _mvcReferenceAssemblies;
+                    }
+
```

Comment:
why do you care about caching and statics here - this is test code.. just write it in a simple and clear way.